docs/project/contribution_tools.md

```diff
@@ -268,3 +269,46 @@ Dwarf Error: DW_FORM_strx1 found in non-DWO CU
 
 It means that the version of GDB used is too old, and does not support the DWARF
 v5 format.
+
+### Debugging on MacOS
+
+Getting Bazel to build debuggable binaries on MacOS requires passing a few extra
+options while building:
+
+```shell
+bazel build --spawn_strategy=local -c dbg //explorer --apple_generate_dsym
```

Comment:
So, first off, super excited to have these steps. I don't think I would have ever figured out what steps to try here. Hat's off to you. All of the instructions worked perfectly for me.

I was really curious how this is so very broken on macOS, so I did a bunch of digging. I think the best explanation is scattered in some (but not all) of the comments on a Bazel bug and summarized pretty well here:
https://github.com/bazelbuild/bazel/issues/2537#issuecomment-449089673

Might be worth linking to that bug from the docs?

After reading that and trying a few things, I don't think you need `--apple_generate_dsym` here. As far as I can tell, it's not generating dsyms in the build, and that flag influences the Apple build rules (objc_binary and such) rather than the C/C++ rules.

The local spawn strategy is the key thing, because *without* dsyms, the linker encodes the paths to the `.o` files during the link, and the `.o` files to the source files during the compile. When sandboxed, none of these paths work after the build. But by disabling the sandbox, the paths computed are all stable and work.

And I think the issue with the build cache is because which path is in the `.o` file isn't a function of anything that keys the build cache.

I'm not sure it's realistic to get the dsyms built until Bazel team fix that bug. But I'm pondering whether we could (a) force macOS builds to use the local strategy somehow so folks don't have to discover this, and (b) cause the relevant paths to be *inputs* to the compiles so that the cache is more reliable. Right now, I think you could get "exciting" situations where two builds in two checkouts could share `.o` files in the cache and end up with LLDB showing source code from the wrong client. Anyways, none of this is blocking -- having these instructions is already a *huge* step forward, so thanks again.

Reply:
Reworked, tested without `--apple_generate_dsym` and added the link to the bug. PTAL